### ISSUE TITLE
underscore in hostname

### ISSUE THREAD

**Issue by Erwane** [layershifter/TLDExtract#25]
Hi,

i've found a weird behavior in domain extraction;

```php
$extract = new Extract();
$result = $extract->parse('dkim._domainkey.phea.fr');
print_r($result->toArray());

$result = $extract->parse('dkim.domainkey.phea.fr');
print_r($result->toArray());
```

**result**
```
Array
(
    [subdomain] => dkim._domainkey.phea
    [hostname] => fr
    [suffix] => 
)
Array
(
    [subdomain] => dkim.domainkey
    [hostname] => phea
    [suffix] => fr
)
```

the problem come from `_` character.

this regex fix the problem
```php
    const HOSTNAME_PATTERN = '#^((?!-)[a-z0-9_-]{0,62}[a-z0-9]\.)+[a-z]{2,63}|[xn\-\-a-z0-9]]{6,63}$#';
```

**Comment by Sm0ke0ut** [comment]
We ran into a similar issue. Would be great if this could be fixed in a relatively short-term. For now we pinned the package onto `1.2.3` as temporary work-around to mitigate the issue.

**Comment by layershifter** [comment]
@Erwane the PR with matching tests will be awesome :+1: Feel free to open it

**Comment by Erwane** [comment]
PR #26

**Comment by layershifter** [comment]
@Erwane thanks, released as `1.2.5`.
